src/runtime/x86.ll

```diff
@@ -131,4 +131,33 @@ define weak_odr <2 x double> @max_f64x2(<2 x double> %a, <2 x double> %b) nounwi
   %c = fcmp olt <2 x double> %a, %b
   %result = select <2 x i1> %c, <2 x double> %b, <2 x double> %a
   ret <2 x double> %result
-}
\ No newline at end of file
+}
+
+; Adapted from asm_cpuid.ll in LLVM source.
+define weak_odr void @halide_x86_cpuid(i32 %fn_id, i32* %out) nounwind uwtable {
```

Comment:
I'd prefer to do it that way, actually, but had some issues with getting cross-compilation right and figured that this was the better way to do assembly. I'll take another stab at it if you prefer that. (As-is, this doesn't build on LLVM3.6 anyway.)